File: monitoring-service/pocket/transaction.go
package pocket

import (
	"fmt"
	"time"
)

const TypeClaim = "pocketcore/claim"
const TypeProof = "pocketcore/proof"

type Transaction struct {
	Hash          string
	Height        uint
	Time          time.Time
	Address       string
	Type          string
	ChainID       string
	NumRelays     uint
	PoktPerRelay  float64
	SessionHeight uint
	ExpireHeight  uint
	AppPubkey     string
	ResultCode    int64
	IsConfirmed   bool
	Reward        Reward
}

func (t Transaction) Chain() (Chain, error) {
	chain, err := ChainFromID(t.ChainID)
	if err != nil {
		return Chain{}, fmt.Errorf("Transaction.Chain: %s", err)
	}

	return chain, nil
}

// PoktAmount is calculated as:
//reward = NUM_RELAYS * RelaysToTokensMultiplier * ((FLOOR/ValidatorStakeFloorMultiplier)/( ServicerStakeWeightMultiplier*ValidatorStakeFloorMultiplier))^(ValidatorStakeFloorMultiplierExponent)
//
func (tx Transaction) PoktAmount() float64 {
	return tx.PoktPerRelay * float64(tx.NumRelays)
}

File: ui/src/types/transaction.ts
import {Chain} from './chain'

export type Transaction = {
    hash: string,
    height: number,
    time: string,
    type: string,
    chain_id: string,
    chain: Chain,
    num_relays: number,
    pokt_per_relay: number,
    pokt_amount: number,
    session_height: number,
    expire_height: number,
    app_pubkey: string,
    is_confirmed?: boolean,
    stake_weight: number,
}